src/Event/Event.jl

```diff
@@ -19,30 +49,44 @@ mutable struct Event{T <: SSDFloat}
     Event{T}() where {T <: SSDFloat} = new{T}(VectorOfArrays(Vector{CartesianPoint{T}}[]), VectorOfArrays(Vector{T}[]), missing, missing)
 end
 
-function Event(location::AbstractCoordinatePoint{T}, energy::RealQuantity = one(T))::Event{T} where {T <: SSDFloat}
+function Event(location::AbstractCoordinatePoint, energy::RealQuantity = 1.0)
+    pt_internal = to_internal_point(location)
+    T = eltype(pt_internal)
     evt = Event{T}()
-    evt.locations = VectorOfArrays([[CartesianPoint(location)]])
-    evt.energies = VectorOfArrays([[T(to_internal_units(energy))]])
+    evt.locations = VectorOfArrays([[pt_internal]])
+    evt.energies  = VectorOfArrays([[T(to_internal_units(energy))]])
     return evt
 end
 
-function Event(locations::Vector{<:AbstractCoordinatePoint{T}}, energies::Vector{<:RealQuantity} = ones(T, length(locations)); max_interaction_distance::Union{<:Real, <:LengthQuantity} = NaN)::Event{T} where {T <: SSDFloat}
+function Event(locations::Vector{<:AbstractCoordinatePoint}, energies::Vector{<:RealQuantity} = fill(1.0, length(locations)); max_interaction_distance::Union{<:Real, <:LengthQuantity} = NaN)
+    pts_internal = to_internal_point.(locations)
+    T = eltype(first(pts_internal))
     d::T = T(to_internal_units(max_interaction_distance))
```

Comment:
Side comment: I fear a bit that determining `T` here (and anywhere else) might create some trouble. If we require `T` of the `Event` to match `T` of the `Simulation` when drifting events, this might not be what we want..

```julia
simulate!(evt::Event{T}, sim::Simulation{T},..)
```
(the `T` need to match and I fear that this implementation might make it a bit unpredictable, especially because the function does not know the `T` of the output `Event{T}` before being called.